library/core/src/ffi/c_str.rs

```diff
@@ -875,4 +880,82 @@ mod verify {
             assert!(c_str.is_safe());
         }
     }
-}
\ No newline at end of file
+
```

Comment:
The stubbing part is a bit unclear : 

Yeah. Since Kani doesnt support external runtime functions, I tried stubbing them, but the functions are in a nested manner (the extern "C" block) , we cannot stub nested functions since its not in the scope and is invoked during runtime. We could try to explicitly mock this function but Im not sure if that is expected.

i.e. the extern "C" block is taken out of this function maintaining functionality of the code OR, we mock the working of the strlen. Reference : https://model-checking.github.io/kani/rfc/rfcs/0002-function-stubbing.html
```
#[inline]
#[unstable(feature = "cstr_internals", issue = "none")]
#[cfg_attr(bootstrap, rustc_const_stable(feature = "const_cstr_from_ptr", since = "1.81.0"))]
#[rustc_allow_const_fn_unstable(const_eval_select)]
const unsafe fn strlen(ptr: *const c_char) -> usize {
    const fn strlen_ct(s: *const c_char) -> usize {
        let mut len = 0;

        // SAFETY: Outer caller has provided a pointer to a valid C string.
        while unsafe { *s.add(len) } != 0 {
            len += 1;
        }

        len
    }

    #[inline]
    fn strlen_rt(s: *const c_char) -> usize {
        extern "C" {
            /// Provided by libc or compiler_builtins.
            fn strlen(s: *const c_char) -> usize;
        }

        // SAFETY: Outer caller has provided a pointer to a valid C string.
        unsafe { strlen(s) }
    }

    intrinsics::const_eval_select((ptr,), strlen_ct, strlen_rt)
}
```

Reply:
@rajathkotyal FYI, strlen was updated:
```rust
const unsafe fn strlen(ptr: *const c_char) -> usize {
    const_eval_select!(
        @capture { s: *const c_char = ptr } -> usize:
        if const {
            let mut len = 0;

            // SAFETY: Outer caller has provided a pointer to a valid C string.
            while unsafe { *s.add(len) } != 0 {
                len += 1;
            }

            len
        } else {
            extern "C" {
                /// Provided by libc or compiler_builtins.
                fn strlen(s: *const c_char) -> usize;
            }

            // SAFETY: Outer caller has provided a pointer to a valid C string.
            unsafe { strlen(s) }
        }
    )
}
```

Reply:
Thank you, this resolves the nested module blocker